homeassistant/components/services.yaml

```diff
@@ -533,3 +533,88 @@ knx:
       data:
         description: KNX data to send
         example: 1
+
+xiaomi_vacuum:
+  send_command:
+    description: Send 'raw' command to the vacuum cleaner.
+
+    fields:
+      host:
+        description: IP address of the vacuum cleaner. Defaults to all hosts.
+        example: '192.168.1.135'
+
+      command:
+        description: Command to execute.
+        example: 'set_dnd_timer'
+
+      params:
+        description: Parameters for the command.
+        example: '[22,0,6,0]'
+
+  set_fanspeed:
+    description: Set the fanspeed of the vacuum cleaner.
+
+    fields:
+      host:
+        description: IP address of the vacuum cleaner. Defaults to all hosts.
+        example: '192.168.1.135'
+
+      fanspeed:
+        description: Fanspeed, between 0 and 100.
+        example: '60'
+
+  remote_control_start:
+    description: Start remote control of the vacuum cleaner. You can then move it with `remote_control_move`, when done call `remote_control_stop`.
+
+    fields:
+      host:
+        description: IP address of the vacuum cleaner. Defaults to all hosts.
+        example: '192.168.1.135'
+
+  remote_control_stop:
+    description: Stop remote control mode of the vacuum cleaner.
+
+    fields:
+      host:
+        description: IP address of the vacuum cleaner. Defaults to all hosts.
+        example: '192.168.1.135'
+
+  remote_control_move:
+    description: Remote control the vacuum cleaner, make sure you first set it in remote control mode with `remote_control_start`.
+
+    fields:
+      host:
+        description: IP address of the vacuum cleaner. Defaults to all hosts.
+        example: '192.168.1.135'
+
+      velocity:
+        description: Speed, between -0.29 and 0.29.
+        example: '0.2'
+
+      rotation:
+        description: Rotation, between -179 degrees and 179 degrees.
+        example: '90'
+
+      duration:
+        description: Duration of the movement?
+        example: '1500'
+
+  remote_control_move_step:
+    description: Remote control the vacuum cleaner, only makes one move and then stops.
+
+    fields:
+      host:
+        description: IP address of the vacuum cleaner. Defaults to all hosts.
+        example: '192.168.1.135'
+
+      velocity:
+        description: Speed, between -0.29 and 0.29.
+        example: '0.2'
+
+      rotation:
+        description: Rotation, between -179 degrees and 179 degrees.
+        example: '90'
+
+      duration:
+        description: Duration of the movement?
```

Comment:
I know, went a little crazy when testing these commands... I'm not sure about them (they where in the original PR from @bramkragten, and they are in your library, so if they are implemented, perhaps some future user will enlighten us with a proper usage.